src/socket.c

```diff
@@ -244,7 +255,8 @@ mrb_basicsocket_getsockname(mrb_state *mrb, mrb_value self)
 static mrb_value
 mrb_basicsocket_getsockopt(mrb_state *mrb, mrb_value self)
 { 
-  int opt, s;
+  char opt[4];
```

Comment:
`sizeof(int)` can be 8 on ILP64 systems.  Why do you change `opt` to an array of char?